device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsConnectionStateCallback.java

```diff
@@ -0,0 +1,19 @@
+package com.microsoft.azure.sdk.iot.device.transport.amqps;
+
+/**
+ * Callback interface for the session handler to callback up to the connection level with status updates
+ */
+public interface AmqpsConnectionStateCallback
+{
+    /**
+     * Called when a device's amqp session has successfully been opened. For multiplexing scenarios, this callback will
+     * be fired once per device.
+     * @param deviceId the id of the device that had its session opened successfully
+     */
+    public void onDeviceSessionOpened(String deviceId);
+
+    /**
+     * Called when the authentication session has successfully been opened. Never called for x509 auth.
```

Comment:
For multiplexing scenarios with multiple devices, we will have 1 auth link associated with the connection, so this will be fired once, correct?

Reply:
Correct, yes. Even for single device connections, that callback only gets fired once.